### DISCUSSION TITLE
Accessing 0x10 Error while invoke non static method

### DISCUSSION THREAD

**Discussion by dvgmdvgm** [vfsfitvnm/frida-il2cpp-bridge#102] [Q&A]
Hello. First of all big thanks for A12 fix.

I want to invoke some method but it's non static.
try to invoke like that : 
```
const classs = Il2Cpp.Domain.assembly("imageName").image.class("className");
classs.method<void>("MethodName").invoke((Il2Cpp.String.from("data"));
```

it gives me this output : 
`Error: access violation accessing 0x10
    at invokeRaw (node_modules/frida-il2cpp-bridge/dist/il2cpp/structs/method.js:168)`

My class have Instance object too.
Maybe i should to invoke my method with Instance object?
How i can achieve it?
Thank you.

**Comment by vfsfitvnm** [comment]
That's a static method, otherwise you would get [another](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/structs/method.ts#L195) error.
Also, it doesn't look like a `frida-il2cpp-bridge` issue, however I cannot give you more info since you didn't provide the signature of the method you are trying to invoke

**Comment by dvgmdvgm** [comment]
@vfsfitvnm i also get this error that you show to me serveral tries before.
`must be invoked throught a Il2Cpp.Object, not a Il2Cpp.Class`
How it can be resolved?
Sorry for distrubing

**Comment by vfsfitvnm** [comment]
You must call that method using a `Il2Cpp.Object`.  E.g:
```cs
class System.String
{
    System.Boolean Contains(System.String value); // 0x0082c664;
    static System.Boolean IsNullOrEmpty(System.String value); // 0x0082c684;
}
```
```ts
const SystemString = Il2Cpp.Image.corlib.class("System.String");
const string = Il2Cpp.String.from("hello");

// Contains is not static, so:
console.log(string.object.method<boolean>("Contains").invoke(Il2Cpp.String.from("el"))); // true

// IsNullOrEmpty is static, so:
console.log(SystemString.method<boolean>("IsNullOrEmpty").invoke(string)); // true
```
The difference is in how you obtain the method: `Contains` is obtained via a `Il2Cpp.Object` (`Il2Cpp.String::object` returns a `Il2Cpp.Object`), `IsNullOrEmpty` is obtained via a `Il2Cpp.Class`.

I don't know how you can retrieve an instance of that class, it really depends on what you are trying to do. You can take a look at the past issues (like #39)

**Reply by qiaozhi-4** [reply]
Hi, I was wondering how to create this step if it's not a String type
`const string = Il2Cpp.String.from("hello");`

**Comment by dvgmdvgm** [comment]
It's all about corlib...
But how about Assembly-CSharp?
Try by this snippet :

```const SystemObject = Il2Cpp.Image.corlib.class("System.Object");

const ClassName1= Il2Cpp.Domain.assembly("Assembly-CSharp").image.class("ClassName");

const Infl = ClassName1.inflate(SystemObject);

const Instance = Infl.new();

log(Instance);

Instance.method("MethodName").invoke(get_Profile("BEST"));


OUTPUT : 
`il2cpp: cannot inflate class ClassName: it has no generic parameters`

**Comment by vfsfitvnm** [comment]
Well it seems there's a bit of confusion here... Why are you trying to inflate a generic class?

Also, did you read the issue I linked?

**Comment by dvgmdvgm** [comment]
working by following snippet
```
const Instance = mod("Assembly-CSharp", "ClassName").method<Il2Cpp.Object>("get_Instance").invoke();
Instance.method("MethodName").invoke(get_Profile("BEST"));
```
It's my fault cause i'm try to invoke method to early. Like immediately after attached to process.
Try to invoke it from another implementation and it's works.. Sorry.

**Reply by kolanski** [reply]
what is mod function?

**Comment by IrvanWijayaSardam** [comment]
looks like my question is belongs to here, i hope somebody can help, i still learning, how do i call <Awake>b__25_7 ? i've got this message 

`il2cpp: cannot invoke non-static method <Awake>b__25_7 as it must be invoked throught a Il2Cpp.Object, not a Il2Cpp.Class
    at invoke (il2cpp/structs/method.ts:214)
    at <anonymous> (src/index.ts:30)
    at call (native)
    at <anonymous> (il2cpp/structs/method.ts:328)`
    
    
    this is my frida script
    `import "frida-il2cpp-bridge";

console.log("Frida loaded successfully !!");

Il2Cpp.perform(function () {
        

    const AssemblyCSharp = Il2Cpp.domain.assembly("Assembly-CSharp").image;

    const TableManager = AssemblyCSharp.class("TableManager");
    const LobbyMenu = AssemblyCSharp.class("LobbyMenu");

    const OnBackBtn = TableManager.method("OnBackBtn");
    const AwakeB257 = LobbyMenu.method("<Awake>b__25_7");

    OnBackBtn.implementation = function (unused) {
        console.log("Back Button Hooked !!");
        AwakeB257.invoke();
    };

});
`

and this is the awake function from dnSpy
`	private void <Awake>b__25_7() {}
`
